src/main/java/uk/ac/ebi/atlas/experiments/ScExperimentTrader.java

```diff
@@ -0,0 +1,57 @@
+package uk.ac.ebi.atlas.experiments;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.model.experiment.Experiment;
+import uk.ac.ebi.atlas.trader.ExperimentTrader;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+@Component
+public class ScExperimentTrader {
```

Comment:
A trader is a class that exchanges (i.e. trades) one thing for another. In the case of the `experimentTrader` it trades experiment accessions for experiment objects. This class does something else, and should be renamed (and the DAO class with it too).

Reply:
Renamed class accordingly.

Reply:
The point stands: your class is not a trader.

Reply:
Moved the logic to experiment service class after our discussion yesterday.